src/nginx/grpc_server_call.cc

```diff
@@ -691,6 +694,37 @@ bool NgxEspGrpcServerCall::TryReadDownstreamMessage() {
   return true;
 }
 
+// Allocate the chain link and buffer.
+ngx_chain_t *NgxEspGrpcServerCall::AllocNgxBufChain(size_t buflen) {
+  ngx_chain_t *cl = ngx_chain_get_free_buf(r_->pool, &buf_free_);
+  if (!cl) {
+    return nullptr;
+  }
+  ngx_buf_t *buf = cl->buf;
+
+  if (buflen > size_t(buf->end - buf->start)) {
+    // Abandon the small buf, allocate a bigger one.
+    buf = ngx_create_temp_buf(r_->pool, buflen);
+    if (!buf) {
+      return nullptr;
+    }
+    cl->buf = buf;
+  }
+
```

Comment:
My understanding of this function is:
Part 1:  get a buf.  First call ngx_chain_get_free_buf to get a free buf, kind of reuse mechanism.  Then if the free buf cannot satisfy the length requirement, create a new temp buf.  
            the original logic is just create a new temp buf directly for all new response. 
            This part seems not direct related to the memory leak.
Part 2:  recycle the buf. This part is the actual part solving the memory leak. 

Is this correct?